include/deal.II/base/aligned_vector.h

```diff
@@ -465,9 +465,215 @@ class AlignedVector
 
 private:
   /**
-   * Pointer to actual data array.
+   * A class that is used as the "deleter" for a `std::unique_ptr` object that
+   * AlignedVector uses to store the memory used for the elements.
+   *
+   * There are two ways the AlignedVector class can handle memory:
+   * - Allocation via `new[]` in reserve() where we call `posix_memalign()`
+   *   to obtain a chunk of memory and then do placement-`new` to initialize
+   *   memory. If this is what we have done, then we need to call the
+   *   destructors of the currently active elements by hand, and then
+   *   call `std::free()` to return memory. In order to call the destructors
+   *   of currently used elements, the deleter object needs to have access
+   *   to the owning `AlignedVector` object to know which of the allocated
+   *   elements are currently actually used.
+   * - We have called `replicate_across_communicator()`, in which case the
+   *   elements have been moved into a memory "window" managed by MPI.
+   *   In that case, one process (the root process of an MPI communicator
+   *   that ties together all processes on one node) needs to call the
+   *   destructor of all elements in this shared memory space, and then
+   *   all processes on that communicator need to first destroy the
+   *   MPI window object, and then the MPI communicator for the shared
+   *   memory node's processes. In this approach, we need to store the
+   *   following data: A pointer to the owning AlignedVector object to know
+   *   which elements need to be destroyed, copies of the MPI window
+   *   and communicator objects, and a couple of ancillary pieces of data.
+   *
+   * In a previous incarnation of this functionality, these deleters were
+   * encoded into lambda functions that captures whatever information was
+   * necessary, and that were then stored in a `std::function` object that
```

Comment:
This is an internal class, so I thought I'd talk about the design decisions that led to what is in the current patch. I put this snippet in there because the *usual* design uses lambda functions/`std::function` objects as deleters: We use this in a number of other places. So my goal was to *contrast* what we do here with the usual approach. I could have phrased that as "The usual approach to this functionality..." instead of "In a previous incarnation of this function...". Would you prefer this?

Reply:
Yes, that makes more sense to me.